Iteration 1 of Savanna

22DPVEmer opened the pull request:
Implemented a console-based Savanna game with a split architecture (Console frontend and GameEngine backend) where players can add Antelopes ('A') and Lions ('L') to a field. The game features intelligent animal behavior where Antelopes try to escape from Lions using their superior speed while Lions hunt Antelopes using better vision range, with all movements and interactions managed by a tick-based game engine. Currently, the grid is hardcoded and the animal attributes can not be changed from the user side.


1. Image of the game screen in its starting state:
![image](https://github.com/user-attachments/assets/fc8bed07-612f-4a5e-b4f0-1eea153a2102)
2. Image of the active game screen with lions and antelopes:
![image](https://github.com/user-attachments/assets/d6a80906-7903-49a7-b848-67664d976bed)

22DPVEmer committed:
Iteration 1: Console game

22DPVEmer committed:
Merge branch 'Iteration_1' of https://github.com/22DPVEmer/ViestursEmerbergs_Savanna into temp_merge_branch

22DPVEmer committed:
Merging Iteration 1: Console game implementation

   - Added console application frontend with Code Library backend
   - Implemented animal movement and interaction logic
   - Added Lion and Antelope classes with chase/avoid behavior
   - Implemented game field and animal management

22DPVEmer committed:
Removed unused method and constant.

22DPVEmer committed:
Remove tracked files that should be ignored

sreeharihm commented on Savanna.ConsoleApp/Program.cs:
Its better to move Display , program run out of this file

22DPVEmer replied:
I moved them into separate files.

sreeharihm commented on Savanna.GameEngine/Models/Antelope.cs:
```diff
@@ -0,0 +1,157 @@
+            do
+            {
+                int direction = Random.Next(8); // 8 possible directions
+                int dx = ((direction + 1) % 3 - 1) * Speed; // -1, 0, or 1 times Speed
```
Move these hard coded values to a constant fields

22DPVEmer replied:
Moved them to the GameConstants.

22DPVEmer committed:
Refactor game initialization and movement constants

- Extracted game initialization logic to a new GameRunner class
- Added movement-related constants in GameConstants
- Updated Antelope movement logic to use new constants
- Simplified Program.cs main method

sreeharihm reviewed (approved):
Looks good!!
